src/module.js

```diff
@@ -66,12 +66,19 @@ function module_derive(injects, injectModule) {
     injectByAlias.set(inject.alias, inject);
   });
   var copy = new Module(this._runtime);
+  copy._ready = false;
   Promise.resolve().then(() => this._copy(copy, injectByAlias, injectModule, new Map));
   return copy;
 }
 
 function module_copy(copy, injectByAlias, injectModule, map) {
   map.set(this, copy);
+  if (this._ready === false) {
+    console.warn("unable to derive uninitialized module!");
+    return copy;
+  }
+  copy._ready = false;
```

Comment:
This feels slightly funky to me — as `copy._ready` should already be set before `module_copy` is called. (And we can probably also use a slightly more descriptive flag...)

We could DRY it up by replacing our `new Module` calls with:

```js
function module_for_copy(runtime) {
  const module = new Module(runtime);
  module._initializing = true;
  return module;
}
```